spec/TestResultsQualityGate/TestResultsQualityGateService.spec.ts

```diff
@@ -0,0 +1,483 @@
+        type = createQualityGate(settings, mockWebApi);
+        expect(type.type).toEqual(TypeEnum.TOTAL_EXECUTED_TESTS);
+        settings.type = 'newlyFailed';
+        type = createQualityGate(settings, mockWebApi);
+        expect(type.type).toEqual(TypeEnum.NEWLY_FAILED_TESTS);
+
+        settings.type = 'TOTALPASSED';
+        type = createQualityGate(settings, mockWebApi);
+        expect(type.type).toEqual(TypeEnum.TOTAL_PASSED_TESTS);
+        settings.type = 'TOTALFAILED';
+        type = createQualityGate(settings, mockWebApi);
+        expect(type.type).toEqual(TypeEnum.TOTAL_FAILED_TESTS);
+        settings.type = 'TOTALEXECUTED';
+        type = createQualityGate(settings, mockWebApi);
+        expect(type.type).toEqual(TypeEnum.TOTAL_EXECUTED_TESTS);
+        settings.type = 'NEWLYFAILED';
+        type = createQualityGate(settings, mockWebApi);
+        expect(type.type).toEqual(TypeEnum.NEWLY_FAILED_TESTS);
+    });
+
+    it('Setting Quality Gate buildStatus', () => {
+        settings.buildStatus = 'unknown';
+        let unknown = createQualityGate(settings, mockWebApi);
+        expect(unknown.buildStatus).toEqual(BuildStatusEnum.FAILED);
+        expect(tl.warning).toHaveBeenCalledWith('Invalid value for \'buildStatus\': unknown, using default value \'failed\'');
+
+        settings.buildStatus = 'unstable';
+        let unstable = createQualityGate(settings, mockWebApi);
+        expect(unstable.buildStatus).toEqual(BuildStatusEnum.UNSTABLE);
+        settings.buildStatus = 'failed';
+        let failed = createQualityGate(settings, mockWebApi);
+        expect(failed.buildStatus).toEqual(BuildStatusEnum.FAILED);
+
+        settings.buildStatus = 'Unstable';
+        unstable = createQualityGate(settings, mockWebApi);
+        expect(unstable.buildStatus).toEqual(BuildStatusEnum.UNSTABLE);
+        settings.buildStatus = 'Failed';
+        failed = createQualityGate(settings, mockWebApi);
+        expect(failed.buildStatus).toEqual(BuildStatusEnum.FAILED);
+    });
+
+    describe('When evaluate quality gate and', () => {
+        let config: QualityGateTestConfig = {
+            pipelines: [],
+            builds: [],
+            currentTestResults: [],
+            referenceTestResults: []
+        }
+        let setUpQualityGate = (config: QualityGateTestConfig) => {
+            let qualityGateService = createQualityGate(settings, mockWebApi);
+            spyOn(qualityGateService.apiClient, 'getPipelinesByName').and.returnValue(Promise.resolve(config.pipelines));
+            spyOn(qualityGateService.apiClient, 'getBuildsOfPipelineById').and.returnValue(Promise.resolve(config.builds));
+            spyOn(qualityGateService.apiClient, 'getTestResultsByBuildId').and.returnValues(Promise.resolve(config.currentTestResults), Promise.resolve(config.referenceTestResults));
+            return qualityGateService;
+        }
+        let compareMarkDown = (expectedReportPath: string) => {
```

Comment:
Can we refactor the code under ’specs' to have a common utils type with methods like this one?